examples/add_pedestal_noise_template.xml

```diff
@@ -0,0 +1,130 @@
+            spikeLimit="${RemoveSpikes.spikeLimit}"
+            topSlopeLimit="${RemoveSpikes.topSlopeLimit}"
+            maxSpikeLength="${RemoveSpikes.maxSpikeLength}"
+        />
+        <fact.datacorrection.InterpolateTimeSeries
+                calibService="calibService"
+                dataKey="DataCalibrated"
+                dataOutputKey="DataCalibrated"
+        />
+        
+        
+    <!-- Sample Pedestal Event and calibrate it -->	
+        <fact.utils.SamplePedestalEvent prependKey="LONS_" noiseDatabase="${noiseDB}" dataFolder="${dataFolder}"
+                                        dbBinningKey="Zd" itemBinningKey="MPointingPos.fZd" binning="${noiseBinning}"
+                                        samplingTryKey="Tries"
+        />
+        
+        <fact.features.UnixTimeUTC2DateTime inputKey="LONS_UnixTimeUTC" outputKey="LONS_timestamp" />
+    
+        <fact.datacorrection.DrsCalibration
+                drsKey="LONS_drspath"
+                key="LONS_Data"
+                outputKey="LONS_DataCalibrated"
+                startCellKey="LONS_StartCellData"
+        />
+
+        <fact.datacorrection.PatchJumpRemoval
+                dataKey="LONS_DataCalibrated"
+                outputKey="LONS_DataCalibrated"
+                prevEventsKey="LONS_prevEvents"
+                startCellKey="LONS_StartCellData"
+                unixTimeKey="LONS_UnixTimeUTC"
+                jumpLimit="${PatchJumpRemoval.jumpLimit}"
+        />
+
+         <fact.datacorrection.RemoveSpikes
+                dataKey="LONS_DataCalibrated"
+                outputKey="LONS_DataCalibrated"
+                startCellKey="LONS_StartCellData"
+                leftBorder="${RemoveSpikes.leftBorder}"
+                spikeLimit="${RemoveSpikes.spikeLimit}"
+                topSlopeLimit="${RemoveSpikes.topSlopeLimit}"
+                maxSpikeLength="${RemoveSpikes.maxSpikeLength}"
+        />
+
+        <fact.datacorrection.DrsTimeCalibration
+                startCellKey = "LONS_StartCellData"
+                dataKey      = "LONS_DataCalibrated"
+                outputKey    = "LONS_DataCalibrated"
+        />
+
+        <fact.datacorrection.InterpolateTimeSeries
+                calibService="calibService"
+                dataKey="LONS_DataCalibrated"
+                dataOutputKey="LONS_DataCalibrated"
+                badPixelKey="LONS_badPixel"
+                timeStampKey="LONS_timestamp"
+        />
+
+        <fact.utils.CombineDataArrays firstArrayKey="DataCalibrated" secondArrayKey="LONS_DataCalibrated"
+                                      outputKey="DataCalibrated" op="add"/>
+        
+         <fact.datacorrection.DrsCalibration
+                key="DataCalibrated"
+                outputKey="Data"
+                reverse="true"
+        />
+        <fact.utils.Remapping
+                key="Data"
+                outputKey="Data"
+                reverse="True"
+        />
+        
+        <stream.script.JavaScript>
```

Comment:
Don't we maybe want to set the creator in the processor that actually applies the noise instead of using inline javascript in the xml?

Reply:
The apply doesn't really happens until latter in the `CombineDataArrays` and not in the `SamplePedestalEvent` processor. Or to put it more simply the whole set of processors create the finished product, that is why it is set here explicitly.

We could also say that the 'SamplePedestalEvent' processor changes the creator but on its own it just adds additional stuff to the data item and doesn't change anything in the original item.

Reply:
> The apply doesn't really happens until latter in the CombineDataArrays and not in the SamplePedestalEvent processor. Or to put it more simply the whole set of processors create the finished product, that is why it is set here explicitly.

Is there any usecase of using just one of the processors? If not, maybe we should incroparate this into only one processor.

Reply:
The problem with applying it into a single processor is, that the two processors are not side by side. Because we first need to calibrate the event before we can combine them, so there are other processor in between and putting all of that into a single processor is not the nicest thing to do.

I think the best solution will be to export the whole thing into an module that can kinda act like a single processor, with the parameters set trough properties (I don't think modules have parameters, right?). That way users can use it as a kinda processor but we don't get all the additional code if we make a mega processor (It also allows for easy changes to the calibration part and so on). Because if you think about it this whole thing is like a processor made out of processors.

Reply:
Yes I agree to Michael putting everything in one processor creates a lot copy code (e.g. Drs_Calibrations) that we should avoid. Thus I up for keeping the process in several containers and would vote for adding a module like the cleaning.xml